### PR TITLE
Avoid column-wrapping array values in debug strings

### LINKED ISSUE #869: Prevent trace content wrapping
Hi there!

I really enjoy OpenFisca, but I recently encountered an issue.

### Here is what I did:

`(.venv) ➜  mes-aides-ui git:(update_of) ✗ openfisca-run-test ../openfisca-paris/tests/unittests/aspeh.yaml --country-package openfisca_france --extensions openfisca_paris -v > aspeh
`

### Here is what I expected to happen:

To have an unwrapped indented trace that can easily be manipulated in Sublime Text for instance.
Sans titre

That issue is generated by the explicit enum values (changed from numbers to enum value, many thanks!).

### Here is what actually happened:

Long lines are wrapped and the trace is not easily explorable.

### Here is data (or links to it) that can help you reproduce this issue:

```
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-4.4.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/thomas/repos/openfisca-paris
collected 4 items

tests/unittests/aspeh.yaml Computation log:
  paris_logement_aspeh<2016-02> >> [0.]
    parisien<2016-02> >> [ True]
    paris_enfant_handicape<2016-02> >> [False False False False  True]
      handicap<2016-02> >> [False False False False  True]
      paris_enfant<2016-02> >> [False False  True  True  True]
        est_enfant_dans_famille<2016-02> >> [False False  True  True  True]
        enfant_place<2016-02> >> [False False False False False]
        enfant_a_charge<2016> >> [False False  True  True  True]
          age<2016-01> >> [55 55  5  7  9]
          handicap<2016-12> >> [False False False False False]
    paris_base_ressources_couple<2016-01> >> [5344.6836]
      en_couple<2016-01> >> [ True]
        nb_parents<2016-01> >> [2]
      paris_base_ressources_i<2016-01> >> [3100. 1400.    0.    0.    0.]
        ass<2016-01> >> [0. 0. 0. 0. 0.]
          ass_base_ressources<2016-01> >> [0.]
            ass_base_ressources_individu<2016-01> >> [0. 0. 0. 0. 0.]
              salaire_imposable<2015-01> >> [0. 0. 0. 0. 0.]
                salaire_de_base<2015-01> >> [0. 0. 0. 0. 0.]
                primes_salaires<2015-01> >> [0. 0. 0. 0. 0.]
                primes_fonction_publique<2015-01> >> [0. 0. 0. 0. 0.]
                  categorie_salarie<2015-01> >> ['prive_non_cadre' 'prive_non_cadre' 'prive_non_cadre' 'prive_non_cadre'
 'prive_non_cadre']
                  traitement_indiciaire_brut<2015-01> >> [0. 0. 0. 0. 0.]
                indemnite_residence<2015-01> >> [0. 0. 0. 0. 0.]
                  traitement_indiciaire_brut<2015-01> >> [0. 0. 0. 0. 0.]
                  salaire_de_base<2015-01> >> [0. 0. 0. 0. 0.]
                  categorie_salarie<2015-01> >> ['prive_non_cadre' 'prive_non_cadre' 'prive_non_cadre' 'prive_non_cadre'
 'prive_non_cadre']
```

## Context

I identify more as a:
- Developer _(I create tools that use the existing OpenFisca code)_.

### PR DESCRIPTION
**Pull request by Morendil**
#### Bug fix

- Avoid column-wrapping array values in debug strings

Fixes #869

### CODE REVIEW

File: openfisca_core/indexed_enums.py [hunk 1]
```diff
@@ -131,4 +131,4 @@ def __repr__(self):
         return '{}({})'.format(self.__class__.__name__, str(self.decode()))
 
     def __str__(self):
-        return str(self.decode_to_str())
+        return np.array2string(self.decode_to_str(),max_line_width=float("inf"))
```
Comment (fpagnoux) [inline openfisca_core/indexed_enums.py]: > Regular numpy arrays are wrapped when stringified. Why should EnumArray behave differently? I could see some issues when large population simulations.

Comment (Morendil) [reply openfisca_core/indexed_enums.py]: See #869 referenced in the PR description: we don't want arrays wrapped.

This is a design flaw in Numpy that this PR works around; arrays and strings are abstract data types, converting between array and string representation of the same underlying data should not be tied to a notion of "column width of the terminal" which is not only medium-specific but has no necessary relation to the terminal where the information will be viewed.

Admittedly this PR perhaps needs a unit test or two, I'm not actually sure the specific code for EnumArray is needed for the computation log case as they are a subclass of Numpy arrays.

**Comment by Morendil** [comment]
PR fixed and ready for review.

**Comment by sandcha** [comment]
What I understand from the issue is that, instead of having an output similar to this for enums:

```sh
    paris_base_ressources_i<2019-01> >> [529. 145.  75.]
      ...
      tns_auto_entrepreneur_benefice<2019-01> >> [0. 0. 0.]
        tns_auto_entrepreneur_type_activite<2019-01> >> ['achat_revente' 'achat_revente' 
'achat_revente']
        tns_auto_entrepreneur_chiffre_affaires<2019-01> >> [0. 0. 0.]
```

We would like to have `['achat_revente' 'achat_revente' 'achat_revente']` on the same line whatever the list length is.
@guillett Could you add an example of the expected output? 🙂 

> I couldn't reproduce the issue with `aspeh` unit test (seems to have changed since) but I had a similar output with: `openfisca-run-test ../openfisca-paris/tests/unittests/paris.yaml --country-package openfisca_france --extensions openfisca_paris -v > paris`

**Comment by guillett** [comment]
Yes @sandcha,  variables should use a single line.

```
    paris_base_ressources_i<2019-01> >> [529. 145.  75.]
      ...
      tns_auto_entrepreneur_benefice<2019-01> >> [0. 0. 0.]
        tns_auto_entrepreneur_type_activite<2019-01> >> ['achat_revente' 'achat_revente' 'achat_revente']
        tns_auto_entrepreneur_chiffre_affaires<2019-01> >> [0. 0. 0.]
```

**Pull request merged**